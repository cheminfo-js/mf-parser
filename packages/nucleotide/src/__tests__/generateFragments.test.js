'use strict';

const { generateFragments } = require('..');

describe('test generate fragments', () => {
  it('nucleotide to sequence of OHDampDtmpDcmpH', () => {
    let fragments = generateFragments('HODampDtmpDcmpH', { a: true });
    expect(fragments).toStrictEqual(['HODampO-1H-1$a1', 'HODampDtmpO-1H-1$a2']);
  });
  it('nucleotide to sequence of HODampDtmpDcmpH', () => {
    let fragments = generateFragments('HODampDtmpDcmpH', { a: true, ab: true });
    expect(fragments).toStrictEqual([
      'HODampO-1H-1$a1',
      'HC5H6O4P$a1 - B',
      'HODampDtmpO-1H-1$a2',
      'HODampC5H6O4P$a2 - B'
    ]);
  });
  it('nucleotide to sequence of HODampDtmpDcmpH and many fragments', () => {
    let fragments = generateFragments('HODampDtmpDcmpH', {
      a: true,
      ab: true,
      b: true,
      c: true,
      d: true,
      w: true,
      x: true,
      y: true,
      z: true
    });
    expect(fragments).toStrictEqual([
      'HODampO-1H-1$a1',
      'HC5H6O4P$a1 - B',
      'HODamp$bH-11',
      'HODampPO2$c1',
      'HODampPO3H2$d1',
      'HODcmpH$w1',
      'H-2DcmpH$x1',
      'O-2H-1P-1DcmpH$y1',
      'O-3H-4P-1DcmpH$z1',
      'HODampDtmpO-1H-1$a2',
      'HODampC5H6O4P$a2 - B',
      'HODampDtmp$bH-12',
      'HODampDtmpPO2$c2',
      'HODampDtmpPO3H2$d2',
      'HODtmpDcmpH$w2',
      'H-2DtmpDcmpH$x2',
      'O-2H-1P-1DtmpDcmpH$y2',
      'O-3H-4P-1DtmpDcmpH$z2'
    ]);
  });

  it('nucleotide HODampDtmpDcmpH internal aby fragments', () => {
    let fragments = generateFragments('HODampDtmpDcmpDgmpDampH', {
      abw: true
    });
    expect(fragments).toStrictEqual([
      'HODtmpC5H6O4P$B2:B3',
      'HODtmpDcmpC5H6O4P$B2:B4',
      'HODcmpC5H6O4P$B3:B4'
    ]);
  });
});
