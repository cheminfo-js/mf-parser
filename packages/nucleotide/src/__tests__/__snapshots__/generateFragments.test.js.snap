// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test generate fragments nucleotide fragmentation a of HODadeDtmpDcmpH 1`] = `
Array [
  "HODadeO-1H-1$a1",
  "HODadeDtmpO-1H-1$a2",
]
`;

exports[`test generate fragments nucleotide fragmentation ab of HODadeDtmpDcmpH 1`] = `
Array [
  "HODadeC5H6O4P$a2 - B",
]
`;

exports[`test generate fragments nucleotide fragmentation y of HODadeDtmpDcmpH 1`] = `
Array [
  "O-2H-1P-1DcmpH$y1",
  "O-2H-1P-1DtmpDcmpH$y2",
]
`;

exports[`test generate fragments nucleotide to sequence of HODampDtmpDcmpH 1`] = `
Array [
  "HODampO-1H-1$a1",
  "HODampDtmpO-1H-1$a2",
]
`;

exports[`test generate fragments nucleotide to sequence of HODampDtmpDcmpH 2`] = `
Array [
  "HODampO-1H-1$a1",
  "HODampDtmpO-1H-1$a2",
  "HODampC5H6O4P$a2 - B",
]
`;

exports[`test generate fragments nucleotide to sequence of HODampDtmpDcmpH 3`] = `
Array [
  "HODampO-1H-1$a1",
  "HODamp$b1",
  "HODampPO2H$c1",
  "HODampPO3H$d1",
  "ODcmpH$w1",
  "DcmpH$x1",
  "O-2H-1P-1DcmpH$y1",
  "O-3H-2P-1DcmpH$z1",
  "HODampDtmpO-1H-1$a2",
  "HODampC5H6O4P$a2 - B",
  "HODampDtmp$b2",
  "HODampDtmpPO2H$c2",
  "HODampDtmpPO3H$d2",
  "ODtmpDcmpH$w2",
  "DtmpDcmpH$x2",
  "O-2H-1P-1DtmpDcmpH$y2",
  "O-3H-2P-1DtmpDcmpH$z2",
]
`;
